File: RemoteAccessJava/app/src/main/java/com/lokesh/remoteaccessjava/MainActivity.java
package com.lokesh.remoteaccessjava;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.ActivityCompat;
import androidx.core.content.ContextCompat;

import android.Manifest;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.util.Log;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Log.d("adnan", "on create");
        checkForSmsReceivePermissions();

        Bundle extras = getIntent().getExtras();
        if (extras != null) {
            String address = extras.getString("address");
            String message = extras.getString("message");
            Log.d("adnan", extras.getString("address"));
            Log.d("adnan", extras.getString("message"));
            TextView addressField = (TextView) findViewById(R.id.address);
            TextView messageField = (TextView) findViewById(R.id.message);
            addressField.setText(address);
            Log.d("adnan", address);
            messageField.setText(message);
            Log.d("adnan", message);
        }

    }

    void checkForSmsReceivePermissions(){
        // Check if App already has permissions for receiving SMS
        if(ContextCompat.checkSelfPermission(getBaseContext(), "android.permission.RECEIVE_SMS") == PackageManager.PERMISSION_GRANTED) {
            // App has permissions to listen incoming SMS messages
            Log.d("adnan", "checkForSmsReceivePermissions: Allowed");
        } else {
            // App don't have permissions to listen incoming SMS messages
            Log.d("adnan", "checkForSmsReceivePermissions: Denied");

            // Request permissions from user
            ActivityCompat.requestPermissions(this, new String[] {Manifest.permission.RECEIVE_SMS}, 43391);
        }
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
        if(requestCode == 43391){
            if(grantResults.length>0 && grantResults[0] == PackageManager.PERMISSION_GRANTED){
                Log.d("adnan", "Sms Receive Permissions granted");
            } else {
                Log.d("adnan", "Sms Receive Permissions denied");
            }
        }
    }
}

File: RemoteAccess/app/src/main/java/com/lokesh/remoteaccess/MainActivity.kt
package com.lokesh.remoteaccess

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    //Function called when the SetAccessKeyButton is clicked
    fun SetAccessKey(view: View) {
        val TAG = "MyActivity"
        Log.d(TAG, "Access key is : "+accessKey_edittext.editableText.toString())
        val sharedPrefe = this.getPreferences(Context.MODE_PRIVATE) ?: return
        with(sharedPrefe.edit()){
            putString("access_key",accessKey_edittext.toString())
            commit()
            Log.d(TAG, "Access key saved")
        }
    }

    fun getData(view: View) {
        val sharedPref = this.getPreferences(Context.MODE_PRIVATE) ?: return
        val str_name = sharedPref.getString("access_key", "")
        Toast.makeText(this, "$str_name ", Toast.LENGTH_LONG).show()

    }

}